[Docs] Add user personas (#268)
OpenAssetIO/OpenAssetIO: contributing/USER_PERSONAS.md

# OpenAssetIO User Personas

> Note: This page is a WIP, please add/suggest omissions through PR

As a project, the majority of OpenAssetIO users are software developers.
We count a "user" as someone who interacts with the code base and/or
distribution, rather than with any higher-order functionality exposed by
an application integrating the API.

> -   [Traditional DCC tool author](#traditional-commercial-dcc-tool-author)

Reviewer: Maybe just "Commercial"? Easy to understand what's intended.

Author: Good call, thanks :)

> -   [Propriatory DCC tool author](#proprietary-dcc-tool-author)

Reviewer: ```suggestion
-   [Proprietary DCC tool author](#proprietary-dcc-tool-author)
```

> -   [Propriatory DAM/AMS/PT system author](#proprietary-damamspt-system-author)

Reviewer: ```suggestion
-   [Proprietary DAM/AMS/PT system author](#proprietary-damamspt-system-author)
```

-   [Commercial DAM/AMS/PT system author](#commercial-damamspt-system-author)
-   [Pipeline Technical Director (TD)](#pipeline-technical-director-td)

## Traditional commercial DCC tool author

> DCC: Digital Content Creation, usually used to refer to tools such as
> Houdini, Nuke, Katana, Mari, Maya, Photoshop, ZBrush, et. al. that are
> used within the Media Creation sector to produce digital equivalents
> of the traditional mediums for making 2D/3D artwork/models. Known as a
> "host" in OpenAssetIO speak.

These developers predominantly work as part of a larger team,

> maintaining desktop applications that typically run on bare metal close
> to the user. Most of these developers work for an ISV (Independent

Reviewer: ```suggestion
maintaining desktop applications that typically run on desktop operating systems. These applications themselves often act as a closed platform for end user and plugin developers. Most of these developers work for an ISV (Independent
```

Software Vendor) who license their products through a perpetual licence

-   annual maintenance or subscription model. Releases of their tools are
    usually a mix of iterative refinement and headline feature development

>     to drive the sales cycle, as such, release windows are subject to

Reviewer: ```suggestion
    to drive the sales cycle. As such, release windows are subject to
```

> truth for the state and location of data within a media creation
> production pipeline. Known as a "manager" in OpenAssetIO speak.

These developers work as part of an in-house team delivering the
production critical systems used within Media Creation environments such
as visual effects houses and animation studios. Similarly to the
proprietary tool developer, the main drive is to deliver specific
internal deadlines and workflow requirements.

Historically, the pipeline supported by these systems has been the
"secret sauce" forming the competitive advantage for any particular
business, by allowing forecasting and delivery to be as efficient as
possible. As such, the systems developed are often tightly coupled to a
particular companies business processes, and require deep integration
into artist tooling.

These systems are often a mix of centralized and decentralized
databases, distributed processing management, web services,
in-application plugins and extensions and miscellaneous glue code in a
variety of languages.

-   **OpenAssetIO API alignment:** Manager
-   **Languages:** JavaScript, C++, Python, Lua, CSS, HTML, \*SQL,

>     GraphQL

Reviewer: ```suggestion
    GraphQL. Rust is anticipated as a target in coming years.
```

> facilities.

These developers work as part of an in-house team to solve technical
challenges that arise during the delivery of a project. They may well
also be part of a team with the Proprietary DAM/AMS/PT system authors
described above. Though often more focused towards project-specific
deliverables, they also contribute to the general maintenance and
development of the pipeline itself.

When working under the broader remit, and on more immediate solutions,
their work may tend towards higher-level APIs and scripts over low-level
frameworks and shared library development. This means that from the
perspective of OpenAssetIO they may quickly jump from the host API to
the manager API. E.g, using the host API to quickly build scripts to
deal with data or asset management tasks. Maybe to support an ad-hoc
workflow required by a particular task that doesn't fit into the general
pipeline.

> There is debate as to whether they would use an abstraction such as
> OpenAssetIO over the first-class native SDK of the pipeline, but as it
> is a possibility, we should consider it.

-   **OpenAssetIO API alignment:** Manager, Host

> -   **Languages:** Python, Lua, CSS, HTML, \*SQL, GraphQL

Reviewer: ```suggestion
-   **Languages:** Python, Lua, CSS, HTML, \*SQL, GraphQL. Rust has started to pick up momentum in some studios.
```

# OpenAssetIO User Personas

> Note: This page is a WIP, please add/suggest omissions through PR

As a project, the majority of OpenAssetIO users are software developers.
We count a "user" as someone who interacts with the code base and/or
distribution, rather than with any higher-order functionality exposed by
an application integrating the API.

> -   [Commercial DCC tool author](#commercial-dcc-tool-author)

Author: 🤔 I'd love to just use that, but do you think it's obvious enough? I quite liked having one in-line in the docs to form some last-ditch attempt to avoid Wall-of-Text-Syndrome™

deal with data or asset management tasks. Maybe to support an ad-hoc
workflow required by a particular task that doesn't fit into the general
pipeline.

> There is debate as to whether they would use an abstraction such as
> OpenAssetIO over the first-class native SDK of the pipeline, but as it
> is a possibility, we should consider it.

-   **OpenAssetIO API alignment:** Manager, Host
-   **Languages:** Python, Lua, CSS, HTML, \*SQL, GraphQL. Rust has
    started to pick up momentum in some studios.
-   **Platforms:** Linux, Windows, macOS
-   **Frameworks:** Qt, pyside, database technologies
-   **Codebase age:** From short-lived to long term depending on
    specific tasks.
-   **Ability to adopt new language standards:** Mixed, often
    constrained by technology choices that exist within the pipeline
    itself, however new solutions may be free to innovate as required to
    solve a specific need.

Interested in:

-   Simple, easy to us, stable, robust, documented APIs.
-   Ways to solve a problem quickly.
-   Interoperability between tools.
-   Automation.
-   Availability of tools via common scripting languages.

Uses OpenAssetIO to:

> -   Quickly build tools and scripts that interact with asset management

Reviewer 2: [question] Is there a use case of bridging multiple asset management systems, e.g. for subcontracting, or combined use of commercial and in-house systems? 

I guess this would be covered by your more general statement, but from my naive perspective some more specific examples would be nice.

Author: Great point. Added [here](https://github.com/TheFoundryVisionmongers/OpenAssetIO/pull/268/commits/4bd79aaee88d509f9459b4500967d2cfdd989f11)
